yt_dlp/extractor/opencast.py

```diff
@@ -0,0 +1,201 @@
+                            hs-harz\.opencast\.uni-halle\.de|
+                            videocampus\.urz\.uni-leipzig\.de|
+                            media\.uct\.ac\.za|
+                            vid\.igb\.illinois\.edu|
+                            cursosabertos\.c3sl\.ufpr\.br|
+                            mcmedia\.missioncollege\.org|
+                            clases\.odon\.edu\.uy
+                        )'''
+    _UUID_RE = r'[\da-fA-F]{8}-[\da-fA-F]{4}-[\da-fA-F]{4}-[\da-fA-F]{4}-[\da-fA-F]{12}'
+
+    def _call_api(self, host, video_id, note=None, errnote=None, fatal=True):
+        return self._download_json(self._API_BASE % (host, video_id), video_id, note=note, errnote=errnote, fatal=fatal)
+
+    def _parse_mediapackage(self, video):
+        tracks = try_get(video, lambda x: x['media']['track'])
+
+        video_id = video.get('id')
+
+        formats = []
+        for track in tracks:
+            href = track['url']
+            ext = determine_ext(href, None)
+            track_obj = {'url': href}
+
+            transport = track.get('transport')
+
+            if transport == 'DASH' or ext == 'mpd':
+                formats.extend(self._extract_mpd_formats_and_subtitles(href, video_id, mpd_id='dash', fatal=False))
+            elif transport == 'HLS' or ext == 'm3u8':
+                formats.extend(
+                    self._extract_m3u8_formats_and_subtitles(href, video_id, m3u8_id='hls', entry_protocol='m3u8_native', fatal=False)
+                )
+            elif transport == 'HDS' or ext == 'f4m':
+                formats.extend(self._extract_f4m_formats(href, video_id, f4m_id='hds', fatal=False))
+            elif transport == 'SMOOTH':
+                formats.extend(self._extract_ism_formats(href, video_id, ism_id='smooth', fatal=False))
+            elif ext == 'smil':
+                formats.extend(self._extract_smil_formats(href, video_id, fatal=False))
+            else:
+                if transport is not None:
+                    track_obj.update({'format_note': track.get('transport')})
+                    if transport == 'RTMP':
+                        m_obj = re.search(r'(?:rtmp://[^/]+/(?P<app>[^/]+))/(?P<ext>.+):(?P<playpath>.+)', href)
+                        if not m_obj:
+                            continue
+                        track_obj.update(
+                            {
+                                'app': m_obj.group('app'),
+                                'ext': m_obj.group('ext'),
+                                'play_path': m_obj.group('ext') + ':' + m_obj.group('playpath'),
+                                'rtmp_live': True,
+                                'preference': -2,
+                            }
+                        )
+
+                audio_info = track.get('audio')
+                if audio_info is not None:
+                    track_obj.update({
+                        'abr': int_or_none(audio_info.get('bitrate'), 1000),
```

Comment:
is it really safe to use `1000` as default here?

Reply:
that's `scale`, not `default`